workflow/controller/operator_test.go

```diff
@@ -4684,8 +4684,8 @@ func TestTemplateTimeoutDuration(t *testing.T) {
 		makePodsPhase(t, apiv1.PodPending, controller.kubeclientset, wf.ObjectMeta.Namespace)
 		woc.operate()
 
-		assert.Equal(t, wfv1.NodeFailed, woc.wf.Status.Phase)
-		assert.Equal(t, wfv1.NodeFailed, woc.wf.Status.Nodes.FindByDisplayName("hello-world-dag").Phase)
+		assert.Equal(t, wfv1.NodeError, woc.wf.Status.Phase)
```

Comment:
I'm not sure what is correct here. This code change is exercised by this test. The steps test marks the node as `Error` but the workflows as `Failed`. I think the steps should also mark as `Error` - or both should be `Failed`.

@sarabala1979 thoughts? @jessesuen

Reply:
Oh - for context - this is the new timeout code @sarabala1979 just merged.

Reply:
I think we need to be clear when something is an error and when something is a failure. I'm a bit unclear on this looking at the code.

Reply:
Testcase is wrong. I will fix it. It should check the failed for the step which timed out

Reply:
I've made a change to your implementation to return a specific `ErrTimeout` that allows us to switch based on the result.

However, I'm still not sure if the node and workflow should be failed or errored.